### PR TITLE
Add the capability to verify more integrity factors

### PR DESCRIPTION
**Pull request by jacksenechal**
In order to verify more than just the table name matching, we also need
row data. We'll be able to do things like match the fields and values in
the SQL to the tuples in the row data.

### CODE REVIEW

File: src/main/scala/changestream/ChangeStreamEventListener.scala [hunk 1]
```diff
@@ -254,7 +254,9 @@ object ChangeStreamEventListener extends EventListener {
       case Some(e: MutationEvent)     =>
         ChangeStreamEventListener.inFlightIncrement(e.rows.length)
         val nextPosition = getNextPosition
-        IntegrityVerifier.matchesTableAndSql(e.tableName, e.sql.getOrElse(""), nextPosition, "ChangestreamEventListener")
+        val rowData = getRowData(e)
```
Comment (cpalafox) [inline src/main/scala/changestream/ChangeStreamEventListener.scala:257]: this bit is failing due to type mismatch, given e is a `MutationEvent`, while RowData needs a `MutationWithInfo`

what we can do here is.

```
val mutationWithInfo = MutationWithInfo(e, nextPosition)
val rowData = getRowData(mutationWithInfo)

IntegrityVerifier.verify(e.tableName, e.sql.getOrElse(""), nextPosition, rowData, "ChangestreamEventListener")
transactionActor ! mutationWithInfo
```

File: src/main/scala/changestream/IntegrityVerifier.scala [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
 import org.slf4j.LoggerFactory
+import spray.json.JsValue
+
+import scala.collection.immutable.ListMap
 
 object IntegrityVerifier {
   protected val log = LoggerFactory.getLogger(getClass)
   protected val sqlTableNameRegex = raw"((?i)insert into|update|delete from) (`?\w+`?\.)?`?(\w+)`?".r.unanchored
 
+  def verify(tableName: String, sql: String, position: String, rowData: Seq[ListMap[String, JsValue]], source: String = ""): Boolean = {
```
Comment (cpalafox) [inline src/main/scala/changestream/IntegrityVerifier.scala:12]: we provide rowData here but we don't use it? is it just it being WIP?

File: src/main/scala/changestream/RowDataExtractor.scala [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import scala.collection.immutable.ListMap
+
+import spray.json._
+import DefaultJsonProtocol._
+
+object RowDataExtractor {
+  protected val log = LoggerFactory.getLogger(getClass)
+  val dateFormatter: DateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+  val textFieldSizeLimit: Integer = 16000 // 16KB
+  val truncationText: String = "...[TRUNCATED]"
```
Comment (cpalafox) [inline src/main/scala/changestream/RowDataExtractor.scala:20]: I think the JsonFormatterActorSpec was using this data, so we'll need to change that

**Review by cpalafox — CHANGES_REQUESTED** [review]
Saw some places we need to change stuff at to fix build problems

**Comment by jacksenechal** [comment]
Archiving

**Pull request closed without merging**